src/framework/mpas_string_utils.F

```diff
@@ -0,0 +1,54 @@
+! Copyright (c) 2021-2023 The University Corporation for Atmospheric Research (UCAR).
```

Comment:
Apologies for the misleading suggestion to follow the copyright statement from `mpas_halo.F` file. In this case, I think the copyright year can be 2023, since this is a new file only introduced this year.